passage/src/main/java/id/passage/android/CurrentUser.kt

```
@@ -226,12 +137,101 @@ final class PassageUser private constructor(
+     * @throws PassageUserException If an error occurs during the deletion process
+     */
+    suspend fun deleteSocialConnection(socialConnectionType: SocialConnection) {
+        val currentUserAPI = CurrentuserAPI(Passage.BASE_PATH)
+        try {
+            return currentUserAPI.deleteCurrentuserSocialConnection(Passage.appId, socialConnectionType)
+        } catch (e: Exception) {
+            throw PassageUserException.convert(e)
+        }
+    }
+
+    /**
+     * Retrieve Current User Metadata
+     *
+     * Fetches the metadata associated with the current user. This method requires that the user is authenticated via a bearer token.
+     * The metadata contains various attributes associated with the user, which can be used for user-specific operations.
+     * @return Metadata? containing the current user's metadata, null if retrieval fails
+     */
+    suspend fun metadata(): Metadata? {
+        val currentUserAPI = CurrentuserAPI(Passage.BASE_PATH)
+        return try {
+            currentUserAPI.getCurrentuserMetadata(Passage.appId)
+        } catch (e: Exception) {
+            null
+        }
+    }
+
+    /**
+     * Update User Metadata
+     *
+     * Updates the metadata associated with the current user. This method requires that the user is authenticated via a bearer token.
+     * The `metadata` parameter should contain the new metadata that will be associated with the user.
+     * @param metadata The Metadata object containing the updated user information
+     * @return CurrentUserInfo? containing the updated user information, null if the update fails
+     */
+    suspend fun updateMetadata(metadata: Metadata): CurrentUserInfo? {
+        val currentUserAPI = CurrentuserAPI(Passage.BASE_PATH)
+        val request = UpdateMetadataRequest(userMetadata = metadata.userMetadata)
+        return try {
+            currentUserAPI.updateCurrentuserMetadata(Passage.appId, request).user
+        } catch (e: Exception) {
+            null
```

I think we want to throw an exception here.